Week-3/Exercise_3.1/memorize.js

```diff
@@ -0,0 +1,32 @@
+// function to muliply given arguments
+// example 1,2,3 ==> 1*2*3
+
+function multiply(...args) {
+  return args.reduce((a, b) => a * b);
+}
+
+// memorization function
+function memorize(func) {
+  // store to cache the data
+  const store = new Map();
+  return function (...args) {
+    // creating key to store the result
+    // example 1,2,3 ==> 1-2-3
+    const keyId = args.join("-");
```

Comment:
Instead of using the key as a string with delimeter as '-' , could have thought of better thing like Array of sorted numbers as string which will be key.

So that when the similar arguments were provided in different order as well we need not compute the output